compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -70,20 +70,15 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 _ => (),
             }
 
+            self.create_def_if_needed_for(e);
             let hir_id = self.lower_node_id(e.id);
             self.lower_attrs(hir_id, &e.attrs);
 
             let kind = match &e.kind {
                 ExprKind::Array(exprs) => hir::ExprKind::Array(self.lower_exprs(exprs)),
                 ExprKind::ConstBlock(c) => {
                     let c = self.with_new_scopes(c.value.span, |this| {
-                        let def_id = this.create_def(
```

Comment:
If the NodeId has already been lowered, `create_def` should either ICE or register it correctly.

Reply:
What do you want to see changed here? I'm a little confused by the purpose of this comment haha